app/models/task.py

```diff
@@ -1,5 +1,42 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import ForeignKey
+from typing import Optional
 from ..db import db
+from datetime import datetime
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from .goal import Goal
+
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str] = mapped_column(nullable=False)
+    description: Mapped[str] = mapped_column(nullable=False)
+    completed_at: Mapped[datetime] = mapped_column(nullable=True)
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+
+    def is_complete(self):
+        if not self.completed_at:
+            return False
+        return True
+
+    def to_dict(self):
+        task_dict = {
+            "id": self.id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.is_complete()
+        }
+        if self.goal_id is not None:
+            task_dict["goal_id"] = self.goal_id
```

Comment:
Instead of checking that `self.goal_id` is not None, you can check that it's a Truthy value:
```suggestion
        if self.goal_id:
            task_dict["goal_id"] = self.goal_id
```